dts/bindings/reserved-memory/reserved-memory.yaml

```diff
@@ -0,0 +1,23 @@
+description: Reserved memory
+
+compatible: "reserved-memory"
+
+include:
+  - name: base.yaml
+    property-allowlist: ['#address-cells', '#size-cells']
+
+child-binding:
+    description: Regions
+    properties:
+       label:
+          type: string
+          required: false
+          description: Human readable string describing the device (used as device_get_binding() argument)
+       reg:
+          type: array
+          description: register space
+          required: true
+       linker-section:
```

Comment:
My initial reaction is that `linker-section` should not be here.

Reply:
I'd prefer we drop `linker-section` from the devicetree until we have a view of this across all of devicetree and zephyr.

Reply:
linker-section` was added to guarantee backward compatibility with the reserved-memory nodes already present in Zephyr (if this property is not present, then nothing is actually done with the region). Can you expand on why you don't like this?

Reply:
What do you mean by backward compatibility with the existing usage?

I don't like this as I dont think the idea of `linker-section` in the devicetree has been completely thought out.  Why don't we have `linker-section` for sram or memory nodes?  Why only `reserved-memory`?  I'd prefer we have the users that want such linker sections to explicitly add them to their linker scripts and can use DT as a source of the "defines".

Reply:
ok